### PR TITLE
kanha: init at 0.1.2

### PR DESCRIPTION
**Pull request by pwnwriter**
## Description of changes

Kanha is a web-app pentesting suite written in rust !

https://github.com/pwnwriter/kanha

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ka/kanha/package.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      with darwin.apple_sdk.frameworks;
+      [
+        Security
+        CoreFoundation
+        SystemConfiguration
+      ]
+    );
+
+  meta = {
+    description = "A web-app pentesting suite written in rust";
```
Comment (Sigmanificient) [inline pkgs/by-name/ka/kanha/package.nix:35]:
```suggestion
    description = "Web-app pentesting suite written in rust";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ka/kanha/package.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  lib,
+  rustPlatform,
+  fetchCrate,
+  stdenv,
+  darwin,
+  pkgs,
```
Comment (eclairevoyant) [inline pkgs/by-name/ka/kanha/package.nix:7]:
```suggestion
  openssl,
```

never pass in `pkgs`

File: pkgs/by-name/ka/kanha/package.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchCrate {
+    inherit version;
+    pname = "kanha";
+    hash = "sha256-ftTmYCkra3x/oDgGJ2WSf6yLeKXkwLJXhjuBdv7fVLY=";
+  };
+
+  cargoHash = "sha256-kjHLc+qWo5dB4qbdlIWzk/pjpghRaDcX/7kkjEM219c=";
+
+  buildInputs =
+    with pkgs;
```
Comment (eclairevoyant) [inline pkgs/by-name/ka/kanha/package.nix:23]:
```suggestion
```

Comment (pwnwriter) [reply pkgs/by-name/ka/kanha/package.nix]: I've made the required changes and force pushed with the same commit, Could you now please review one?

File: pkgs/by-name/ka/kanha/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (drupol) [inline pkgs/by-name/ka/kanha/package.nix:1]: Did you format your file with `nixfmt-rfc-style` ?

Comment (eclairevoyant) [reply pkgs/by-name/ka/kanha/package.nix:1]: based on the "peculiarities" I've noticed that the formatter has, it seems they did.

**Pull request merged**